src/main/java/com/exasol/errorreporting/ErrorMessageBuilder.java

```diff
@@ -0,0 +1,158 @@
+package com.exasol.errorreporting;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Builder for Exasol error messages.
+ */
+class ErrorMessageBuilder {
+    private static final Pattern PLACEHOLDER_PATTERN = Pattern.compile("\\{\\{([^\\}]*)\\}\\}");
+    private final String errorCode;
+    private final StringBuilder messageBuilder = new StringBuilder();
+    private final List<String> mitigations = new ArrayList<>();
+    private final Map<String, String> parameterMapping = new HashMap<>();
+
+    /**
+     * Create a new instance of
+     * 
+     * @param errorCode Exasol error code
+     */
+    ErrorMessageBuilder(final String errorCode) {
+        this.errorCode = errorCode;
+    }
+
+    /**
+     * Add exception message.
+     * <p>
+     * If this method is called multiple times, the message is appended.
+     * </p>
+     * 
+     * @param message exception message
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder message(final String message) {
+        this.messageBuilder.append(message);
+        return this;
+    }
+
+    /**
+     * Add a parameter. This method quotes the parameter.
+     * <p>
+     * You can use the parameter in message and mitigation using {@code {{parameter}}}.
+     * </p>
+     * 
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder parameter(final String placeholder, final Object value) {
+        return this.unquotedParameter(placeholder, Quoter.quoteObject(value));
+    }
+
+    /**
+     * Add a parameter. This method quotes the parameter.
+     * <p>
+     * You can use the parameter in message and mitigation using {@code {{parameter}}}.
+     * </p>
+     * 
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @param description description for the error catalog
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder parameter(final String placeholder, final Object value, final String description) {
+        return this.parameter(placeholder, value);
```

Comment:
The description is not used cause if's only for the crawler, right?

Reply:
yes